### PR TITLE
Process awareness and termination

### PR DESCRIPTION
**Pull request by nalbion**
- `development/parse_task.prompt` now includes:

> the following processes are running: 
> - "app" - (npm start)
> - "mongo" (mongod)

- add a `kill_process` option in `IMPLEMENT_TASK` functions
- wait for `timeout` or `success_message` when starting long-running processes. I've been having `npm start` crash, so as long as that happens within timeout the LLM will know

- Refactored `Project.get_full_file_path()` as it was failing on `path: '/', file: 'package.json'`
- Test on Windows & Ubuntu

### CODE REVIEW

**Review by dtleal — APPROVED** [review]

File: pilot/helpers/Project.py [hunk 1]
```diff
@@ -212,88 +214,44 @@ def save_file(self, data):
...
-        (File.insert(app=self.app, path=data['path'], name=data['name'], full_path=data['full_path'])
+        (File.insert(app=self.app, path=path, name=name, full_path=full_path)
          .on_conflict(
             conflict_target=[File.app, File.name, File.path],
             preserve=[],
-            update={'name': data['name'], 'path': data['path'], 'full_path': data['full_path']})
+            update={'name': name, 'path': path, 'full_path': full_path})
          .execute())
 
     def get_full_file_path(self, file_path: str, file_name: str) -> Tuple[str, str]:
```
Comment (LeonOstrez) [inline pilot/helpers/Project.py:230]: this function is not working properly for many test cases on mac, lets remove changes made to this function from this PR and do it in separate PR just for this function refactoring once it is finished
